backend/tracim_backend/lib/utils/utils.py

```diff
@@ -499,11 +502,42 @@ def get_build_version(path: str) -> str:
         return repo.head.object.hexsha[:10]
 
 
-def validate_page_token(page_token: str):
+def validate_page_token(page_token: str) -> None:
     # INFO - G.M - 2020-07-23 - Because they lack an explicit message, we catch exceptions from
     # the unserialize_bookmark method of sqlakeyset and re-raise them with an explicit message.
     # See https://github.com/djrobstep/sqlakeyset/issues/34
     try:
         unserialize_bookmark(page_token)
     except Exception as e:
         raise ValidationError('Page token "{}" is not a valid page token'.format(page_token)) from e
+
+
+def validate_simple_dict(dict_: typing.Dict) -> None:
+    """
+    A simple dict validator with:
+    key as 0-9a-zA-Z-_. based string
+    value as either: string, int, bool, float types
+    """
+    for key in dict_.keys():
+        if not isinstance(key, str):
+            raise ValidationError('Dictionary key "{}" is not a string'.format(key))
+        regex_validator = Regexp(regex=(re.compile("^[0-9a-zA-Z-_.]+$")))
+        try:
+            regex_validator(key)
+        except ValidationError as exc:
+            raise ValidationError(
+                'Dictionary key "{}" incorrect : {}'.format(key, str(exc))
+            ) from exc
+
+    # INFO - G.M - We do consider float is the type used for float conversion,
+    # this may change depending
+    # on how the json parser is configured.
+    float_type = float
+    valid_types = [str, int, bool, float_type]
+    for value in dict_.values():
+        if value and type(value) not in valid_types:
```

Comment:
Could be easier by using `isinstance`:
```python
for value in dict_.values():
    if not isinstance(value, (str, int, bool, float, type(None))):
        ...
```
And more "user-friendly" by collecting all the invalid keys before raising the exception, eg.
```python
invalid_key_value_pairs = [
    (key, value)
    for key, value in dict_.items()
    if not isinstance(value, (str, int, bool, float, type(None)))
]
if invalid_key_value_pairs:
   raise ValidationError('Only string/number/null values are allowed. Invalid values: {}'.format(invalid_key_value_pairs))
```

Reply:
Oh nice, i didin't know that isinstance could take a list of type as parameters.